lib/soc/interrupt_guard.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+/*
+    Name: interrupt_guard.hpp
+
+    Copyright(c) 2020 Jay Kickliter
+    This code is licensed under MIT license (see LICENSE file for details)
+*/
+
+#include <cstdint>
+
+namespace soc {
+    /**
+     * Provides an RAII style uninterruptible section.
+     *
+     * `interrupt_guard` disables interrupts for its entire lifetime
+     * so care must be taken to limit it's scope when constructed.
+     *
+     * # Example
+     *
+     *     // Interrupts are enabled, but we need to update a
+     *     // non-atomic global variable.
+     *
+     *     {
+     *         interrupt_guard _;
+     *         // interrupts now disabled until next closing brace.
+     *         some_large_global = SomeLargeStruct();
+     *     }   // _'s destructor called here
+     *     // Interrupts are enabled.
+     */
+    class interrupt_guard {
```

Comment:
`Interrupt_guard` (capital **I**) because class can have instance. 
Also filename need to be changed (capital I)